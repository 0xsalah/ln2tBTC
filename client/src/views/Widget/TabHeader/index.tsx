import React, { useState } from 'react'


interface TabHeaderProps {
    defaultTab: string
    onTabClick: (tabId: string) => void
}

const TabHeader: React.FC<TabHeaderProps> = ({ onTabClick, defaultTab }) => {
    const [currentTabId, setCurrentTabId] = useState(defaultTab)

    const handleTabClick = (e: React.MouseEvent<HTMLLIElement>) => {
        const id = e.currentTarget.getAttribute('data-tab-id') ?? defaultTab
        setCurrentTabId(id)
        onTabClick(id)
    }

    return (
        <ul className="tab-nav" data-tabs-nav="">
            <li onClick={handleTabClick} aria-label='swap' data-tab-id="swap" className={currentTabId === 'swap' ? 'is_active' : ''} ><span>Swap</span></li>
            <li onClick={handleTabClick} aria-label='operate' data-tab-id="operate" className={currentTabId === 'operate' ? 'is_active' : ''} ><span>Operate</span></li>
        </ul>
    )
}

export default TabHeader