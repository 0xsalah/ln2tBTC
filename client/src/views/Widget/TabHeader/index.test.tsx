import React from "react";
import { render } from "@testing-library/react";
import TabHeader from ".";

test("snapshot of component rendered with default tab second", () => {
  const component = render(
    <TabHeader onTabClick={() => null} defaultTab="swap" />
  );
  expect(component.baseElement).toMatchSnapshot();
});

test("snapshot of component rendered with default tab first", () => {
  const component = render(
    <TabHeader onTabClick={() => null} defaultTab="operate" />
  );
  expect(component.baseElement).toMatchSnapshot();
});
