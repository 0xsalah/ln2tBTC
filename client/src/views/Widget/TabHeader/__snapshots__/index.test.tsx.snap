// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of component rendered with default tab first 1`] = `
<body>
  <div>
    <ul
      class="tab-nav"
      data-tabs-nav=""
    >
      <li
        aria-label="swap"
        class=""
        data-tab-id="swap"
      >
        <span>
          Swap
        </span>
      </li>
      <li
        aria-label="operate"
        class="is_active"
        data-tab-id="operate"
      >
        <span>
          Operate
        </span>
      </li>
    </ul>
  </div>
</body>
`;

exports[`snapshot of component rendered with default tab second 1`] = `
<body>
  <div>
    <ul
      class="tab-nav"
      data-tabs-nav=""
    >
      <li
        aria-label="swap"
        class="is_active"
        data-tab-id="swap"
      >
        <span>
          Swap
        </span>
      </li>
      <li
        aria-label="operate"
        class=""
        data-tab-id="operate"
      >
        <span>
          Operate
        </span>
      </li>
    </ul>
  </div>
</body>
`;
